File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/domain/store/StoreCreator.java
package com.depromeet.threedollar.domain.domain.store;

import com.depromeet.threedollar.common.docs.ObjectMother;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.util.List;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StoreCreator {

    public static Store create(Long userId, String storeName) {
        return Store.builder()
            .userId(userId)
            .name(storeName)
            .type(StoreType.STORE)
            .latitude(33.0)
            .longitude(124.0)
            .build();
    }

    public static Store createWithDefaultMenu(Long userId, String storeName) {
        Store store = create(userId, storeName);
        store.addMenus(List.of(MenuCreator.create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)));
        return store;
    }

    public static Store create(Long userId, String storeName, double latitude, double longitude) {
        return Store.builder()
            .userId(userId)
            .name(storeName)
            .type(StoreType.STORE)
            .latitude(latitude)
            .longitude(longitude)
            .build();
    }

    public static Store createWithDefaultMenu(Long userId, String storeName, double latitude, double longitude) {
        Store store = create(userId, storeName, latitude, longitude);
        store.addMenus(List.of(MenuCreator.create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)));
        return store;
    }

    public static Store create(Long userId, String storeName, StoreType storeType, double latitude, double longitude) {
        return Store.builder()
            .userId(userId)
            .name(storeName)
            .type(storeType)
            .latitude(latitude)
            .longitude(longitude)
            .build();
    }

    public static Store createWithDefaultMenu(Long userId, String storeName, StoreType storeType, double latitude, double longitude) {
        Store store = create(userId, storeName, storeType, latitude, longitude);
        store.addMenus(List.of(MenuCreator.create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)));
        return store;
    }

    public static Store create(Long userId, String storeName, double latitude, double longitude, double rating) {
        return Store.builder()
            .userId(userId)
            .name(storeName)
            .type(StoreType.STORE)
            .latitude(latitude)
            .longitude(longitude)
            .rating(rating)
            .build();
    }

    public static Store createWithDefaultMenu(Long userId, String storeName, double latitude, double longitude, double rating) {
        Store store = create(userId, storeName, latitude, longitude, rating);
        store.addMenus(List.of(MenuCreator.create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)));
        return store;
    }

    public static Store createDeleted(Long userId, String storeName) {
        Store store = create(userId, storeName);
        store.delete();
        return store;
    }

    public static Store createDeletedWithDefaultMenu(Long userId, String storeName) {
        Store store = create(userId, storeName);
        store.addMenus(List.of(MenuCreator.create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)));
        store.delete();
        return store;
    }

    public static Store createDeleted(Long userId, String storeName, double latitude, double longitude) {
        Store store = create(userId, storeName, latitude, longitude);
        store.delete();
        return store;
    }

    public static Store createDeletedWithDefaultMenu(Long userId, String storeName, double latitude, double longitude) {
        Store store = create(userId, storeName, latitude, longitude);
        store.addMenus(List.of(MenuCreator.create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)));
        store.delete();
        return store;
    }

}

File: threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/store/StoreCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.store

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture
import com.depromeet.threedollar.domain.rds.user.domain.store.MenuCreator.create

@TestFixture
object StoreCreator {

    @JvmOverloads
    @JvmStatic
    fun create(
        userId: Long,
        storeName: String,
        latitude: Double = 36.0,
        longitude: Double = 126.0,
        rating: Double = 0.0
    ): Store {
        return Store.builder()
            .userId(userId)
            .name(storeName)
            .type(StoreType.STORE)
            .latitude(latitude)
            .longitude(longitude)
            .rating(rating)
            .build()
    }

    @JvmOverloads
    @JvmStatic
    fun createWithDefaultMenu(
        userId: Long,
        storeName: String,
        latitude: Double = 34.0,
        longitude: Double = 126.0,
        rating: Double = 0.0,
        storeType: StoreType = StoreType.STORE
    ): Store {
        val store = Store.builder()
            .userId(userId)
            .name(storeName)
            .type(storeType)
            .latitude(latitude)
            .longitude(longitude)
            .rating(rating)
            .build()
        store.addMenus(listOf(create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)))
        return store
    }

    @JvmOverloads
    @JvmStatic
    fun createDeleted(
        userId: Long,
        storeName: String,
        latitude: Double = 34.0,
        longitude: Double = 126.0
    ): Store {
        val store = create(userId, storeName, latitude, longitude)
        store.delete()
        return store
    }

    @JvmOverloads
    @JvmStatic
    fun createDeletedWithDefaultMenu(
        userId: Long,
        storeName: String,
        latitude: Double = 34.0,
        longitude: Double = 126.0
    ): Store {
        val store = createDeleted(userId, storeName, latitude, longitude)
        store.addMenus(listOf(create(store, "메뉴 이름", "메뉴 가격", MenuCategoryType.BUNGEOPPANG)))
        return store
    }

}